distributed/comm/ucx.py

```diff
@@ -160,6 +169,10 @@ def peer_address(self) -> str:
                     np.array([nbytes(f) for f in frames], dtype=np.uint64)
                 )
                 # Send frames
+
+                # It is necessary to first synchronize the device before start sending
```

Comment:
Maybe it's important to be very clear here:

```suggestion
                # It is necessary to first synchronize the default stream before start sending
                # We synchronize the default stream because UCX is not stream-ordered and
                #  the syncing the default stream will wait for other non-blocking CUDA streams.
                # Note this is only sufficient if the data being sent is not currently in use on
                # non-blocking CUDA streams. 
```